internal/service/manifest/orphan/detection_test.go

```diff
@@ -0,0 +1,245 @@
+package orphan_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/service/manifest/orphan"
+)
+
+const (
+	kymaName              = "kyma-123"
+	manifestName          = "kyma-123-template-operator-456"
+	differentManifestName = "kyma-456-template-operator-789"
+)
+
+var errGeneric = errors.New("generic error")
+
+func TestDetectionService_DetectOrphanedManifest_WhenMandatoryModule_ReturnNoError(t *testing.T) {
+	service := orphan.NewDetectionService(&clientStubEmptyModuleStatus{})
+	manifest := &v1beta2.Manifest{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name: manifestName,
+			Labels: map[string]string{
+				shared.IsMandatoryModule: shared.EnableLabelValue,
+			},
+		},
+	}
+
+	err := service.DetectOrphanedManifest(t.Context(), manifest)
+
+	require.NoError(t, err)
+}
+
+func TestDetectionService_DetectOrphanedManifest_WhenDeletionTimestampSet_ReturnNoError(t *testing.T) {
+	service := orphan.NewDetectionService(&clientStubEmptyModuleStatus{})
+	manifest := &v1beta2.Manifest{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name:              manifestName,
+			DeletionTimestamp: &apimetav1.Time{Time: time.Now()},
+		},
+	}
+
+	err := service.DetectOrphanedManifest(t.Context(), manifest)
+
+	require.NoError(t, err)
+}
+
+func TestDetectionService_DetectOrphanedManifest_WhenKymaNotFound_ReturnOrphanedManifestError(t *testing.T) {
+	service := orphan.NewDetectionService(&clientStubKymaNotFound{})
+	manifest := &v1beta2.Manifest{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name: manifestName,
+			Labels: map[string]string{
+				shared.KymaName: kymaName,
+			},
+		},
+	}
+
+	err := service.DetectOrphanedManifest(t.Context(), manifest)
+
+	require.Error(t, err)
+	require.ErrorIs(t, err, orphan.ErrOrphanedManifest)
+}
+
+func TestDetectionService_DetectOrphanedManifest_WhenClientReturnsError_ReturnError(t *testing.T) {
+	service := orphan.NewDetectionService(&clientStubGenericError{})
+	manifest := &v1beta2.Manifest{
```

Comment:
For these cases where we just want to create a default manifest with kyma reference, maybe a test helper func?